sdk/core/core/inc/az_span.h

```diff
@@ -104,9 +140,120 @@ AZ_NODISCARD AZ_INLINE az_span az_span_from_single_item(uint8_t const * ptr) {
 }
 
 AZ_NODISCARD AZ_INLINE az_span az_str_to_span(char const * str) {
-  return (az_span){ .begin = (uint8_t const *)str, .size = strlen(str) };
+  return (az_span){
+    ._internal = { .begin = (uint8_t const *)str, .capacity = strlen(str), .length = strlen(str) }
+  };
+}
+
+/****************** Mutating az_span (used to be az_mut_span in the origins)     ******/
```

Comment:
Remove the part about az_mut_span; it isn't meaningful to other teams/customers.